core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -753,6 +755,20 @@ private[kafka] class Processor(val id: Int,
     connId
   }
 
+  private[network] def enqueueResponse(response: RequestChannel.Response): Unit = {
+    responseQueue.put(response)
+    wakeup()
+  }
+
+  private def receiveResponse(): RequestChannel.Response = {
```

Comment:
Perhaps this method name could be changed to use  `dequeue` or `poll`  instead of `receive`?

Reply:
I chose `dequeueResponse` since we also now have `enqueueResponse`.